R/fotmob_leagues.R

```diff
@@ -61,24 +63,31 @@
     length(league_id)
   )
 
-  if(n_league_urls < n_params) {
+  if(is_league_ids & (n_urls < n_params)) {
     warning(
       sprintf(
-        'Found less leagues than specified (%s < %s).',
-        n_league_urls,
+        "Found less leagues than specified (%s < %s).",
+        n_urls,
         n_params
       )
     )
-  } else if (n_league_urls > n_params) {
+  } else if (is_league_ids & (n_urls > n_params)) {
     warning(
       sprintf(
-        'Found more leagues than specified (%s > %s).',
-        n_league_urls,
+        "Found more leagues than specified (%s > %s).",
+        n_urls,
         n_params
       )
     )
   }
-  league_urls$id
+  urls
+}
+
+.fotmob_get_league_ids <- function(...) {
+  .fotmob_get_league_or_season_ids(
```

Comment:
Given how much we read in from the `worldfootballR_data/` repo, I think we may want to consider adding caching to the package, but this works for now. I think the `{nflreadr}` package may serve as a good example. https://github.com/nflverse/nflreadr/blob/bcbd76142dab099424bc8a8afea4de93bba0fe3e/R/from_url.R#L28-L46